pkg/apis/lighthouse.submariner.io/v2alpha1/types.go

```diff
@@ -69,3 +69,109 @@ type ServiceExportList struct {
 	metav1.ListMeta `json:"metadata"`
 	Items           []ServiceExport `json:"items"`
 }
+
+//+genclient
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient:noStatus
+
+// ServiceImport describes a service imported from clusters in a supercluster.
+type ServiceImport struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// +optional
+	Spec ServiceImportSpec `json:"spec,omitempty"`
+	// +optional
+	Status ServiceImportStatus `json:"status,omitempty"`
+}
+
+// ServiceImportType designates the type of a ServiceImport
+type ServiceImportType string
+
+const (
+	// Services are only accessible via their supercluster IP.
+	SuperclusterIP ServiceImportType = "SuperclusterIP"
+	// Headless services allow backend pods to be addressed directly.
+	Headless ServiceImportType = "Headless"
+)
+
+// ServiceImportSpec describes an imported service and the information necessary to consume it.
+type ServiceImportSpec struct {
+	// +listType=atomic
+	Ports []ServicePort `json:"ports"`
+	// +optional
+	IP string `json:"ip,omitempty"`
+	// +optional
+	Type ServiceImportType `json:"type"`
+	// +optional
+	SessionAffinity corev1.ServiceAffinity `json:"sessionAffinity"`
+	// +optional
+	SessionAffinityConfig *corev1.SessionAffinityConfig `json:"sessionAffinityConfig"`
+	// +optional
+	Items []ClusterServiceInfo `json:"clusterServiceInfo"`
```

Comment:
I would make this part of the status instead, because it's something like the "Clusters" list that they propose -just more extended-, which is status dependent (will vary based on the status of the other clusters, etc..).

In fact you could remove this alltogether and put it into the ClusterStatus like:

```
// ClusterStatus contains service configuration mapped to a specific source cluster
type ClusterStatus struct {
 Cluster string `json:"cluster"`
 ServiceIPs []string `json:"serviceIPs,omitempty"`
}
```

And use "Cluster" instead of ClusterID to adhere to their standard :)

note the "omitempty", if not set, it's not even listed.

Reply:
Moved it to ClusterStatus, it is a good one.